### DISCUSSION TITLE
Linux server

### DISCUSSION THREAD

**Discussion by SandraEliz** [open-sdg/open-sdg#1912] [Q&A] [answered]
Hello

Can you give me more information regarding having the system on a linux server?

Thank you

**Comment by SandraEliz** [comment]
Hello

Does anyone have the platform on a server?

I need guidance.

**Comment by brockfanning** [comment, marked as answer]
@SandraEliz Yes -- although most implementations have used Github for hosting, some have used a Linux server for hosting the production site. We talk about this [here](https://open-sdg.readthedocs.io/en/latest/hosting/linux-server/). Which part are you looking for guidance on?

**Reply by SandraEliz** [reply]
Hello

Is it possible to host the platform in windows with IIS server or flask?

**Reply by brockfanning** [reply]
@SandraEliz Sorry I missed this question. Any web server can be used to host the platform.

**Comment by SandraEliz** [comment]
Hello

I already have a linux server with ubuntu, I want to know what should I do?

Should the repositories be cloned or do you download?
And how would I execute it?

Sorry but I don't have a lot of server experience.

Thanks

**Comment by SandraEliz** [comment]
Hello

Can you contact me with someone who has hosted it on a server so that they can give me guidance?

Thank you

**Comment by brockfanning** [comment]
Hi @SandraEliz sorry - I thought I had replied to this. Since the platform is a "static site", hosting it is very easy. All you need to do is upload the built files to a web server. When using Github for hosting, this is done by the automatic pushing of the built files to the "gh-pages" branch. Github provides an easy way to download the files for a branch by going to:

`https://github.com/[my org]/[my repo]/archive/refs/heads/gh-pages.zip`

So one way to manually deploy to a Linux web server would be download this zip and then upload to the web server.

If you're trying to get the production build, then it's usually the "main" branch, so that would be:

`https://github.com/[my org]/[my org].github.io/archive/refs/heads/main.zip`

Anyway, you can download the zip youself from the Github interface:

![image](https://user-images.githubusercontent.com/1319083/219481659-69039655-9298-4f60-9659-9152157149de.png)

If the site repo is separate from the data repo, then they would both need to be hosted. However in both cases it is simply a "static site" so the hosting is relatively simple. For example there is no need for a database or anything like that.

I will say that if you plan to host with a Linux server, you will need to have a staff member responsible for administering that Linux server.

**Comment by SandraEliz** [comment]
Hello

I already downloaded the files to put them on a server.

I want to have them in a folder inside the server, I already changed all the corresponding routes and the site does not work, it does not take the url that I put.

What am I missing?

Thank you

**Reply by brockfanning** [reply]
@SandraEliz I didn't know exactly what is not working, but it is possible that it has to do with the "baseurl" setting in the Jekyll config file. You would probably need to make sure that the "baseurl" matches whatever folder you would like to serve it from.

For example, if you want your production site to be served from example.com/foo, then in your `_config_prod.yml` you may need to have `baseurl: foo`.

**Reply by SandraEliz** [reply]
The _config.yml files no longer come in the download they comment.

Should I generate that change in the Baseurl before downloading?

**Reply by brockfanning** [reply]
@SandraEliz The _config.yml file and the baseurl setting is used during the building of the site. After the site is built, the _config.yml file serves no purpose. So it is not included in the build.

**Comment by SandraEliz** [comment]
At the moment I am adding site and data in root of the served.

Just change the URLs that point to the github repository.

Now I make changes to the data files and the changes are not reflected, what should be done for this process?

Thank you

**Reply by brockfanning** [reply]
@SandraEliz I'm sorry I can't understand what's going on. If you could provide a link to an example showing the problem that might help.

**Reply by SandraEliz** [reply]
Hello

I already have the site and data on the server

I made changes to an indicator's csv file, but the change is not reflected on the site.

![image](https://user-images.githubusercontent.com/117297044/220382489-dc87e4bc-154c-4bef-9974-839dabe819cf.png)


I saw that there are json files and also both the csv and json are duplicated in several folders.

![image](https://user-images.githubusercontent.com/117297044/220382921-fd5dc1c3-f446-42c7-88e7-b5324d444b5a.png)


In this case, when making changes to any indicator, should they be replicated to all the corresponding files in all the folders where it is duplicated?

Thank you

**Reply by sarah-beasley** [reply]
Hi, are you still having problems with this @SandraEliz ?

**Comment by SandraEliz** [comment]
Hello

Can you tell me which countries have the platform on a server?

Please

Thank you

**Reply by brockfanning** [reply]
@SandraEliz I don't know of any examples of automated deployment to a Linux server, at this time.
